File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/preferences/Preferences.java
package com.themaker.fshmo.klassikaplus.data.preferences;

import android.content.SharedPreferences;
import android.util.Log;
import androidx.annotation.NonNull;
import com.google.gson.Gson;
import io.reactivex.subjects.PublishSubject;

import javax.inject.Inject;

/**
 * Враппер над SharedPreferences.
 */
public class Preferences {

    private static final String KEY_FIRST_RUN = "FIRST_RUN";
    private static final String KEY_MOCK = "MOCK";
    private static final Gson GSON = new Gson();
    private static final String KEY_USER = "USER";
    private static final String KEY_F_USER_ID = "F_USER_ID";
    private static final String KEY_REVISION = "REVISION";
    private static final int DEFAULT_REVISION = 1;
    private static final String TAG = Preferences.class.getName();


    private final SharedPreferences sharedPreferences;

    @Inject
    public Preferences(SharedPreferences preferences) {
        sharedPreferences = preferences;
    }

    public void updateRevision(@NonNull Integer revision) {
        Log.i(TAG, "updateRevision: updating to: " + revision);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt(KEY_REVISION, revision);
        editor.apply();
    }

    public Integer getRevision() {
        Integer rev = sharedPreferences.getInt(KEY_REVISION, DEFAULT_REVISION);
        Log.i(TAG, "getRevision: current revision: " + rev);
        return rev;
    }

    public void setFirstTimeAppLaunch() {
        Log.i(TAG, "setFirstTimeAppLaunch: disabling first launch features");
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putBoolean(KEY_FIRST_RUN, false);
        editor.apply();
    }

    public boolean isFirstTimeAppLaunch() {
        Log.i(TAG, "isFirstTimeAppLaunch: ");
        return sharedPreferences.getBoolean(KEY_FIRST_RUN, true);
    }
}

File: app/src/main/java/com/themaker/fshmo/klassikaplus/data/preferences/Preferences.kt
package com.themaker.fshmo.klassikaplus.data.preferences

import android.content.SharedPreferences
import android.util.Log
import com.google.gson.Gson
import io.reactivex.subjects.PublishSubject

import javax.inject.Inject

/**
 * Враппер над SharedPreferences.
 */
class Preferences @Inject
constructor(private val sharedPreferences: SharedPreferences) {

    val revision: Int
        get() {
            val rev = sharedPreferences.getInt(KEY_REVISION, DEFAULT_REVISION)
            Log.i(TAG, "getRevision: current revision: $rev")
            return rev
        }

    val isFirstTimeAppLaunch: Boolean
        get() {
            Log.i(TAG, "isFirstTimeAppLaunch: ")
            return sharedPreferences.getBoolean(KEY_FIRST_RUN, true)
        }

    fun updateRevision(revision: Int) {
        Log.i(TAG, "updateRevision: updating to: $revision")
        val editor = sharedPreferences.edit()
        editor.putInt(KEY_REVISION, revision)
        editor.apply()
    }

    fun setFirstTimeAppLaunch() {
        Log.i(TAG, "setFirstTimeAppLaunch: disabling first launch features")
        val editor = sharedPreferences.edit()
        editor.putBoolean(KEY_FIRST_RUN, false)
        editor.apply()
    }

    companion object {

        private val KEY_FIRST_RUN = "FIRST_RUN"
        private val KEY_MOCK = "MOCK"
        private val GSON = Gson()
        private val KEY_USER = "USER"
        private val KEY_F_USER_ID = "F_USER_ID"
        private val KEY_REVISION = "REVISION"
        private val DEFAULT_REVISION = 1
        private val TAG = Preferences::class.java.name
    }
}